dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/event/DataRelationsValidator.java

```diff
@@ -322,4 +332,28 @@ private Program getEnrollmentProgramFromEvent(TrackerBundle bundle, Event event)
     }
     return null;
   }
+
+  /**
+   * Given a program with registration, check whether an event in the preheat or payload refers to
+   * enrollment with an existing tracked entity. This should not happen because an enrollment
+   * without a tracked entity is not allowed, see {@link
```

Comment:
I find this sentence a bit confusing.
I think it would be more clear to say something like this:
_It should always find a tracked entity, because an enrollment without a tracked entity is not allowed_